run-make-check.sh

```diff
@@ -76,29 +84,34 @@ function run() {
 
     CMAKE_BUILD_OPTS="-DWITH_FIO=ON -DWITH_GTEST_PARALLEL=ON"
 
-    # Are we in the CI ?
+    echo "Note that this script modifies $USER/.ccache/ccache.conf to ensure maximum performance."
+    echo "If that is not desirable, consider running the script as a different user."
+    echo "Also note that the binaries produced by this script do not contain correct time"
+    echo "and git version information, which may make them unsuitable for debugging."
+    # remove the entropy generated by the date/time embedded in the build
+    CMAKE_BUILD_OPTS="$CMAKE_BUILD_OPTS -D ENABLE_GIT_VERSION=OFF"
+    export SOURCE_DATE_EPOCH=$(date +%D |date -f- +%s)
```

Comment:
If I get this correctly, it will maintain better cache utilization for 24 hours, correct? `SOURCE_DATE_EPOCH` only changes daily, which would then result in a rebuild that won't benefit from any previously cached items.

Reply:
True. That was my first expectation for the CI because of the massive rebuilds.
We could extend it yes, if we set it to everyone.
A week ? A month ?

Reply:
The longer the better. Would there be any major drawback if we set it to `"0"`?

Reply:
Good question, I think at least a week would be useful. Also please note that @robbat2 [raised a good point](https://www.spinics.net/lists/ceph-devel/msg41868.html):

> This may fail in some locales where `%D` is not trivially parsable.
>
> Can I suggest this instead:
> `export SOURCE_DATE_EPOCH=$(date -d 'today 00:00 UTC' +%s)`

Reply:
ccache rotate cache by its own. Maybe 0 is a serious option to optimize at most the cache for everyone.